### PR TITLE
🐛 fix running make generate outside of GOPATH

### LINKED ISSUE #6526: Makefile: make generate-go-openapi is broken if repository is not in GOPATH
**What steps did you take and what happened:**

* Dependabot did create PRs to bump golangci-lint #6522 and the github.com/coredns/corefile-migration dependency #6522
* The follow-up GH Action runs `make generate` [here](https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/workflows/dependabot.yml#L36)

This did results in
* unexpected content for the `zz_generated.openapi.go` files
* as well as having them at a wrong place.

**What did you expect to happen:**

In this case no changes for the `zz_generated.openapi.go` files.

**Anything else you would like to add:**

The reason for that is the dependabot github action [here](https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/workflows/dependabot.yml#L36) does not run `make generate` inside the `GOPATH`.

This results in `make generate-go-openapi` which gets transitively executed to also not run inside the `GOPATH`.

Issue was introduced in #6462 and only affects the generation of openapi code.

For kube-openapi this is still an unresolved issue https://github.com/kubernetes/kube-openapi/issues/233

They did workaround this issue themselves in https://github.com/kubernetes/kube-openapi/issues/225 

[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

### PR DESCRIPTION
**Pull request by schrej**
**What this PR does / why we need it**:
It changes the `generate-go-openapi` and `generate-go-conversion-capd` Makefile targets to produce their output at the correct locations when executed outside of GOPATH.

For the `generate-go-openapi` it's done using a debatable symlink hack.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6526

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -403,20 +404,23 @@ generate-go-conversions-capd: $(CONVERSION_GEN) ## Generate conversions go code
...
 		$(OPENAPI_GEN) \
-			--input-dirs=./$${pkg} \
+			--input-dirs=sigs.k8s.io/cluster-api/$${pkg} \
 			--output-file-base=zz_generated.openapi \
 			--output-package=sigs.k8s.io/cluster-api/$${pkg} \
-			--go-header-file=./hack/boilerplate/boilerplate.generatego.txt; \
-	done
+			--go-header-file=../hack/boilerplate/boilerplate.generatego.txt; \
+	done; \
+	rm sigs.k8s.io/cluster-api
```
Comment (sbueringer) [inline Makefile:423]: probably has to match ./tmp/sigs.k8s.io above

Comment (schrej) [reply Makefile:423]: I'm `cd`ing into that directory, so no

Comment (sbueringer) [reply Makefile:423]: Ups, you're right

**Comment by sbueringer** [comment]
/lgtm

Work on my machine (inside of GOPATH) & CI

**Pull request merged**